pkg/services/ssosettings/ssosettingsimpl/service.go

```diff
@@ -35,29 +37,40 @@ type Service struct {
 	secrets secrets.Service
 	metrics *metrics
 
-	fbStrategies []ssosettings.FallbackStrategy
-	reloadables  map[string]ssosettings.Reloadable
+	fbStrategies  []ssosettings.FallbackStrategy
+	providersList []string
+	reloadables   map[string]ssosettings.Reloadable
 }
 
 func ProvideService(cfg *setting.Cfg, sqlStore db.DB, ac ac.AccessControl,
 	routeRegister routing.RouteRegister, features featuremgmt.FeatureToggles,
-	secrets secrets.Service, usageStats usagestats.Service, registerer prometheus.Registerer) *Service {
-	strategies := []ssosettings.FallbackStrategy{
+	secrets secrets.Service, usageStats usagestats.Service, registerer prometheus.Registerer,
+	settingsProvider setting.Provider, licensing licensing.Licensing) *Service {
+	fbStrategies := []ssosettings.FallbackStrategy{
 		strategies.NewOAuthStrategy(cfg),
-		// register other strategies here, for example SAML
+	}
+
+	providersList := ssosettings.AllOAuthProviders
+	if licensing.FeatureEnabled(social.SAMLProviderName) {
+		fbStrategies = append(fbStrategies, strategies.NewSAMLStrategy(settingsProvider))
+
+		if cfg.SSOSettingsConfigurableProviders[social.SAMLProviderName] {
+			providersList = ssosettings.AllProviders
```

Comment:
It's not very clear from this assignment that `ssosettings.AllProviders` contains the `AllOauthProviders` plus the SAML provider. I'd prefer to move the logic from the ssosettings.go file, line 13:
```append(AllOAuthProviders, social.SAMLProviderName)```
to here because anyway it is only used in this place.

Reply:
Changed as requested.